src/interceptors/WebSocket/implementations/WebSocketClass/WebSocketClassInterceptor.ts

```diff
@@ -0,0 +1,266 @@
+import { invariant } from 'outvariant'
+import type { WebSocketEventsMap } from '../../index'
+import { Interceptor } from '../../../../Interceptor'
+import { WebSocketClassClient } from './WebSocketClassClient'
+import { WebSocketClassServer } from './WebSocketClassServer'
+import type { WebSocketSendData } from '../../WebSocketTransport'
+import { bindEvent } from '../../utils/bindEvent'
+import { WebSocketClassTransport } from './WebSocketClassTransport'
+
+export class WebSocketClassInterceptor extends Interceptor<WebSocketEventsMap> {
+  static symbol = Symbol('websocket-class')
+
+  constructor() {
+    super(WebSocketClassInterceptor.symbol)
+  }
+
+  protected checkEnvironment(): boolean {
+    // Enable this interceptor in any environment
+    // that has a global WebSocket API.
+    return typeof globalThis.WebSocket !== 'undefined'
+  }
+
+  protected setup(): void {
+    const webSocketProxy = Proxy.revocable(globalThis.WebSocket, {
+      construct: (
+        target,
+        args: ConstructorParameters<typeof globalThis.WebSocket>,
+        newTarget
+      ) => {
+        const [url, protocols] = args
+
+        const createConnection = (): WebSocket => {
+          return Reflect.construct(target, args, newTarget)
+        }
+
+        // All WebSocket instances are mocked and don't forward
+        // any events to the original server (no connection established).
+        // To forward the events, the user must use the "server.send()" API.
+        const mockWs = new WebSocketClassOverride(url, protocols)
```

Comment:
All constructed WebSocket instances will point to a "dummy" implementation of the WebSocket class (i.e. mock).

## Reason

`new WebSocket(url)` connections to non-existing addresses throw a special kind of error in the browser that cannot be caught by any means.